Remove Table of Contents from Code Reference

adamwoodnz opened the pull request:
See #76 

This PR removes the Table of Contents from the Code Reference pages of the dev docs, but leaves the jump to links on content headings.

This allows more space in the upper section of the page for source and other important info.

Before:
<img width="1600" alt="Screen Shot 2022-06-17 at 12 03 57 PM" src="https://user-images.githubusercontent.com/1017872/174197243-454cbe61-7153-4734-aa0e-4c8b9ababaa1.png">

After:
<img width="1601" alt="Screen Shot 2022-06-17 at 12 06 00 PM" src="https://user-images.githubusercontent.com/1017872/174197402-3b64488f-951f-4c4d-a49a-2ec4f794f52e.png">

Also included is a change to [improve display of the ToC on small screens as suggested](https://github.com/WordPress/wporg-developer/pull/93#issuecomment-1152525768) by @iandunn :
- Stop hiding it on really small screens (< 480px wide) as this is where it is very useful for jumping down a long page
- Make it full width on small screens (< 43em wide) so that it better handles really long ToC titles, eg. https://developer.wordpress.org/rest-api/frequently-asked-questions/

Before:
<img width="600" alt="Screen Shot 2022-06-21 at 4 30 52 PM" src="https://user-images.githubusercontent.com/1017872/174716463-ab110956-a924-4424-8ee5-9471c44a1c67.png">

After:
<img width="602" alt="Screen Shot 2022-06-21 at 4 31 33 PM" src="https://user-images.githubusercontent.com/1017872/174716546-7ed90cb6-0aa0-4a48-9532-9ec8dd1176de.png">

adamwoodnz committed:
Extract Table of Contents from content and fix it to the right
- On medium screens position it within the header
- On small screens hide it from view

adamwoodnz committed:
Refactor table of contents for backwards compatibility

adamwoodnz committed:
Revert doc change

tellyworth commented:
This looks good to me, I'd like to hear @ryelle's opinion on the a11y impact though.

ryelle commented:
A11y impact is a good question — as far as I know, screen readers have a heading navigation mode, so it wouldn't take anything away there, as the heading structure is still the same. Removing it might even be an improvement for SR users, so it's not just another thing to skip over.

<details>
<summary>Heading structure</summary>

<img width="386" alt="Screen Shot 2022-06-20 at 6 01 05 PM" src="https://user-images.githubusercontent.com/541093/174682724-2f172721-69e1-48f9-bb0f-00b4f7ab54b5.png">

</details>

On the other hand, having a content outline can help with cognitive accessibility. I think this is more an issue on the handbooks, where the content on each page is different and having that outline view can help orient readers. The code reference structure is consistent & predictable, so it's less useful there.

Overall, I think this needs community input before making that decision for handbooks, but I'm 👍🏻  for removing it from the code reference.

adamwoodnz commented:
Thanks @tellyworth and @ryelle,

In terms of getting input on the handbook removal, should we:
a. merge this and see what feedback we get about it being gone from all the handbooks
or 
b. revert removing it from places other than the code reference and instead ask the question somewhere

?

tellyworth commented:
That's a great insight @ryelle about the code reference structure being very predictable.

@adamwoodnz I think in this repo we should remove it from the code reference only and leave it intact in the handbooks. That way we can show people our best representation of how we think things should work.

adamwoodnz committed:
Remove ToC from code reference only

adamwoodnz committed:
Update ToC CSS
- Revert sticky positioning
- Show on very small screens
- Full width on small screens

adamwoodnz committed:
Revert markup changes for sticky ToC on code reference

adamwoodnz commented:
> That's a great insight @ryelle about the code reference structure being very predictable.
> 
> @adamwoodnz I think in this repo we should remove it from the code reference only and leave it intact in the handbooks. That way we can show people our best representation of how we think things should work.

Done!
